File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio3/Java/Libro.java
package Generecidad.Ejercicio3.Java;

public class Libro implements Buscarable {
    private String titulo;
    private String autor;
    private String isbn;
    private double precio;

    public Libro(String titulo, String autor, String isbn, double precio) {
        this.titulo = titulo;
        this.autor = autor;
        this.isbn = isbn;
        this.precio = precio;
    }

    public String getTitulo() {
        return titulo;
    }

    public String getIsbn() {
        return isbn;
    }

    @Override
    public String toString() {
        return String.format("Libro: '%s' por %s (ISBN: %s, Precio: $%.2f)", titulo, autor, isbn, precio);
    }

    @Override
    public boolean esSimilar(String criterio) {
        String lowerCaseCriterio = criterio.toLowerCase();
        return this.titulo.toLowerCase().contains(lowerCaseCriterio) ||
               this.isbn.toLowerCase().contains(lowerCaseCriterio);
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio3/Python/Libro.py

class Libro:
   
    def __init__(self, titulo: str, autor: str, isbn: str, precio: float):
        self.titulo = titulo
        self.autor = autor
        self.isbn = isbn
        self.precio = precio

    def __str__(self):
        return f"Libro: '{self.titulo}' por {self.autor} (ISBN: {self.isbn}, Precio: ${self.precio:.2f})"

    def __repr__(self):
        return self.__str__()

    def es_similar(self, criterio: str) -> bool:
        return criterio.lower() in self.titulo.lower() or criterio.lower() in self.isbn.lower()